listPhoto_module/src/test/java/ru/vyakhirev/listphoto_module/ListPhotosFragmentTest.kt

```diff
@@ -0,0 +1,137 @@
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.ViewActions
+import kotlinx.android.synthetic.main.photo_item.view.*
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.Robolectric
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.android.controller.ActivityController
+import org.robolectric.annotation.Config
+import ru.vyakhirev.core_api.dto.PhotoItem
+import ru.vyakhirev.core_api.dto.PhotoResult
+import ru.vyakhirev.listphoto_module.utils.DataGenerator
+import ru.vyakhirev.listphoto_module.utils.RxRule
+import java.net.HttpURLConnection
+import java.util.concurrent.TimeUnit
+
+@RunWith(RobolectricTestRunner::class)
+@Config(application = TestApp::class, sdk = [Build.VERSION_CODES.O_MR1])
+//@LooperMode(LooperMode.Mode.PAUSED)
+class ListPhotosFragmentTest {
+
+    @get:Rule
+    val rxRule = RxRule()
+
+    var mockWebServer: MockWebServer = MockWebServer()
+
+    private lateinit var dummyList: List<PhotoItem>
+    private lateinit var dummyPhotoResult: PhotoResult
+
+
+    lateinit var fragment: ListPhotosFragment
+
+    lateinit var activityController: ActivityController<FragmentActivity>
+
+    private fun clickOnViewChild(viewId: Int) = object : ViewAction {
+        override fun getConstraints() = null
+
+        override fun getDescription() = "Click on a child view with specified id."
+
+        override fun perform(uiController: UiController, view: View) =
+            ViewActions.click().perform(uiController, view.findViewById<View>(viewId))
+    }
+
+
+    @Before
+    fun setUp() {
+        dummyList = DataGenerator.getDummyGalleryList(35)
+        dummyPhotoResult = PhotoResult(1, 3, 30, 90, dummyList as MutableList<PhotoItem>)
+        val response = MockResponse()
+            .setResponseCode(HttpURLConnection.HTTP_OK)
+            .setBody(getBody())
+            .setBodyDelay(0, TimeUnit.MILLISECONDS)
+
+        mockWebServer.enqueue(response)
+        mockWebServer.start(8080)
+
+        fragment = ListPhotosFragment()
+
+//        launchFragmentInContainer<ListPhotosFragment>()
+        activityController = Robolectric.buildActivity(FragmentActivity::class.java)
+
+        activityController.create().start().resume()
+
+        activityController.get()
+            .supportFragmentManager
+            .beginTransaction()
+            .add(fragment, null)
+            .commit()
+    }
+
+    private fun getBody(): String {
+        return dummyPhotoResult.toString()
+    }
+
+    @After
+    fun clear() {
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    fun `should load photo title into title_TV`() {
+
+        shadowOf(Looper.getMainLooper()).idle()
```

Comment:
А зачем это?

Reply:
Если убрать появляется ошибка:"
java.lang.Exception: Main looper has queued unexecuted runnables. This might be the cause of the test failure. You might need a shadowOf(getMainLooper()).idle() call."